test/test_uri_parser.py

```diff
@@ -553,6 +554,72 @@ def test_port_with_whitespace(self):
         with self.assertRaisesRegex(ValueError, r"Port contains whitespace character: '\\n'"):
             parse_uri("mongodb://localhost:27\n017")
 
+    def test_allow_srv_hosts_with_fewer_than_three_dot_separated_parts(self):
+        with patch("dns.resolver.resolve"):
+            parse_uri("mongodb+srv://localhost/")
+            parse_uri("mongodb+srv://mongo.local/")
+
+    def test_error_when_return_address_does_not_end_with_srv_domain(self):
+        test_cases = [
+            {
+                "query": "_mongodb._tcp.localhost",
+                "mock_target": "localhost.mongodb",
+                "expected_error": "Invalid SRV host",
+            },
+            {
+                "query": "_mongodb._tcp.blogs.mongodb.com",
+                "mock_target": "blogs.evil.com",
+                "expected_error": "Invalid SRV host",
+            },
+            {
+                "query": "_mongodb._tcp.blogs.mongo.local",
+                "mock_target": "test_1.evil.com",
+                "expected_error": "Invalid SRV host",
+            },
+            {
+                "query": "_mongodb._tcp.localhost",
+                "mock_target": "localhost",
+                "expected_error": "Invalid SRV host",
+            },
+            {
+                "query": "_mongodb._tcp.mongo.local",
+                "mock_target": "mongo.local",
+                "expected_error": "Invalid SRV host",
+            },
+            {
+                "query": "_mongodb._tcp.localhost",
+                "mock_target": "test_1.cluster_1localhost",
+                "expected_error": "Invalid SRV host",
+            },
+            {
+                "query": "_mongodb._tcp.mongo.local",
+                "mock_target": "test_1.my_hostmongo.local",
+                "expected_error": "Invalid SRV host",
+            },
+            {
+                "query": "_mongodb._tcp.blogs.mongodb.com",
+                "mock_target": "cluster.testmongodb.com",
+                "expected_error": "Invalid SRV host",
+            },
+        ]
+        for case in test_cases:
+            with patch("dns.resolver.resolve") as mock_resolver:
+
+                def mock_resolve(query, record_type, *args, **kwargs):
+                    mock_srv = MagicMock()
+                    mock_srv.target.to_text.return_value = case["mock_target"]
+                    return [mock_srv]
+
+                mock_resolver.side_effect = mock_resolve
+                domain = case["query"].split("._tcp.")[1]
+                connection_string = f"mongodb+srv://{domain}"
```

Comment:
What do you think about making the test param include the full SRV address:
```
            {
                "uri": "mongodb://localhost",
                "mock_response": "localhost.mongodb",
                "expected_error": "Invalid SRV host",
            },
```
This way we don't need to parse the `_mongodb._tcp` part or dynamically generate a SRV address.

Reply:
Not sure but I think I just fixed the broken test case in 577cd7f5e28dadf595c767debe59d75570ef7ac2. This could go in now AFAICT, so additional changes would be refactors.

Reply:
Oh wait, forgot to check evergreen 😂

Reply:
LGTM